src/services/socket.tsx

```diff
@@ -3,40 +3,87 @@
 import { io } from 'socket.io-client'
 import { useEffect, useState } from 'react'
 
-type ListenerRes = {
-	messageServer: string
-	roomIdServer: string
-	userNameServer: string
+export type ListenerRes = {
+	message: string
+	// roomId: string
+	userName: string
+	timeStamp: string
 }
 
-export interface Event extends ListenerRes {
+type AllUsersPointsRes = {
+	message: ListenerRes[]
+	// roomId: string
+	userName: string
 	timeStamp: string
 }
 
 const socket = io()
 
 export function socketRoomEmitter(
 	emitterName: string,
-	message: string,
-	roomId: string,
+	message: string | ListenerRes[],
 	userName: string,
+	timeStamp: string,
+	roomId: string,
 ) {
-	socket.emit(emitterName, message, roomId, userName)
+	socket.emit(emitterName, message, userName, timeStamp, roomId)
+}
+
+export function useAllUsersPointsListener(listenerName: string) {
+	const [listenerRes, setListenerRes] = useState<AllUsersPointsRes>()
+	useEffect(() => {
+		function onListenerRes(
+			message: ListenerRes[],
+			// roomId: string,
+			userName: string,
+			timeStamp: string,
+		) {
+			setListenerRes({ message, userName, timeStamp })
```

Comment:
question[TS]: Do we really need `userName` and `timeStamp` for every single event? I think it depends on the event, right? Is this you fighting TS too?

Reply:
Nope. I had a use for timestamp on every emit and listen, but I've removed that use. However, I don't wish to remove timestamp until I see how this all comes together. I don't want to add it back in.